usr/share/rear/output/PXE/default/800_copy_to_tftp.sh

```diff
@@ -5,46 +5,43 @@
 # This file is part of Relax-and-Recover, licensed under the GNU General
 # Public License. Refer to the included COPYING for full text of license.
 
-if [[ ! -z "$PXE_TFTP_URL" ]] ; then
-    # E.g. PXE_TFTP_URL=nfs://server/export/nfs/tftpboot
-    local scheme=$( url_scheme $PXE_TFTP_URL )
+if [[ -n "$PXE_TFTP_UPLOAD_URL" ]] ; then
+    # E.g. PXE_TFTP_UPLOAD_URL=nfs://server/export/nfs/tftpboot
+    local scheme=$( url_scheme $PXE_TFTP_UPLOAD_URL )
```

Comment:
```suggestion
    local scheme=$( url_scheme "$PXE_TFTP_UPLOAD_URL" )
```

Reply:
I think quoting is not needed here because
a URL/URI must not contain $IFS characters (like spaces).
Our ulr_* functions are not safe against
characters that are not allowed in a URI/URL.
Permitted characters for values in a URI are
ASCII letters, digits, dot, hyphen, underscore, and tilde
and any other character must be percent-encoded.
See the comment at
https://github.com/rear/rear/blob/rear-2.7/usr/share/rear/lib/global-functions.sh#L261

Reply:
I'm a big fan of defensive programming. The fact that other code parts are not yet safe doesn't mean that we should rely on it. If there was a check somewhere in `prep` that validates the `URL` variables then I'd be also relaxed about it. Because then we protect the code from such things.

Reply:
We rely basically all over the place in ReaR
that config values and values from the system are OK.
E.g. most of our code is not safe against $IFS characters.
So why asking @robertdahlem to fix such things for us?
If we really would be fans of defensive programming
we would have long ago fixed our own things on our own.

FWIW:
I wished I found time work on that but I cannot because
there is always much more other work to do which provides
more value to our users for issues that happen in practice
than making existing code more safe against possible issues
which are not reported in practice by our users.